python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -194,6 +194,31 @@ def __getitem__(self, key: slice) -> Buffer:
             raise ValueError("slice must be C-contiguous")
         return self._getitem(offset=start, size=stop - start)
 
+    def copy(self, deep: bool = True):
+        """
+        Return a copy of Buffer.
+
+        Parameters
+        ----------
+        deep : bool, default True
+            If True, returns a deep copy of the underlying Buffer data.
+            If False, returns a shallow copy of the Buffer pointing to
+            the same underlying data.
+
+        Returns
+        -------
+        Buffer
+        """
+        if deep:
+            with cudf.core.buffer.acquire_spill_lock():
```

Comment:
This feels like a leaky abstraction. We shouldn't be acquiring spill locks in buffer, we should implement `SpillableBuffer.copy` as `with acquire_spill_lock(): return super().copy(...)`.

Reply:
I agree, so concretely, here we should just do:
```
return self._from_device_memory(rmm.DeviceBuffer(ptr=self.get_ptr(mode="read"), size=self.size))
```

And `SpillableBuffer` should implement `copy`:

```
@acquire_spill_lock # or with ...
def copy(self, deep=True):
    return super().copy(deep=deep)
```

Reply:
Done, added a `SpillableBuffer.copy` which holds the lock.